Adding power operator with ^  and code style standardisation

znichola opened the pull request:
[adding power opperator](https://github.com/sophiakoulen/mathematical-expression-parser/commit/174056642e0c640c3000a0bd3293ec5a0b295a97)

znichola committed:
clean up for print tree code

znichola committed:
adding power opperator

znichola committed:
Merge pull request #2 from znichola/dev

Adding

znichola committed:
code styling
adding () to return
only using tabs for indentation
adding newline as last line in file
only using LF for newline character changed from CRLF in some files

znichola committed:
Merge pull request #3 from znichola/dev

code styling

znichola commented:
Standardised the code styling a bit with the latest merge
adding () to return
only using tabs for indentation
adding newline as last line in file
only using LF for newline character changed from CRLF in some files

sophiakoulen commented on print_tree.c:
```diff
@@ -1,143 +1,144 @@
-
-static void	print_token(t_token *tok)
-{
-	// printf("%.5s\n%.5s\n%.5s\n", dict_type(tok), dict_category(tok), dict_name(tok));
```
Please remove commented code if you won't need it

znichola replied:
done!

sophiakoulen commented on parser.c:
```diff
@@ -68,7 +68,7 @@ int	parse_term(char **str, t_tree **left_tree)
 		if (!tok)
 			return (0);
-		if (!(tok->type == operation && (tok->value.c == '*' || tok->value.c == '/')))
+		if (!(tok->type == operation && (tok->value.c == '*' || tok->value.c == '/' || tok->value.c == '^')))
```
The priorities of operations aren't respected. In the case `./eval "4 * 3 ^ 2"`, 3 ^2 should have priority over multiplication. I think power operation cannot be handled in the parse_term function.

znichola replied:
yes, I think it's easier to remove the operator in this pull request and sort it out later.

sophiakoulen commented:
I think it's a great idea to add the power operation. However, that cannot be handled inside the parse_term function like you did. I believe it's best if we rethink the grammar first and then update the code.

znichola committed:
Update evaluate.c

removing power operator

znichola committed:
Update lexer.c

removing power operator

znichola committed:
Update parser.c

removing power operator

znichola committed:
Update print_tree.c

Removing uneeded code that was commented

znichola reviewed (commented):
Removed the uneeded commented code and removed the power operator, to be added later.

znichola committed:
Merge branch 'master' into main

znichola commented:
Ok I think it's possible to merge now, I removed the commented code in print_tree and removed the power operator to be added later when after a re-think.

znichola reviewed (commented):
Ah I had to also review this file. It's done now I think.
